docbook/reference/en/en-US/modules/StringConverter.xml

```diff
@@ -665,5 +665,152 @@ public static class TestResource {
+   or array. For example, a proxy implementing
+   </para>
+   
+<programlisting>
+@Path("path/separator/multi/{p}")
+@GET
+public String pathMultiSeparator(@PathParam("p") @Separator("-") List&lt;String&gt; ss); 
+</programlisting>
+
+   <para>
+   will turn
+   </para>
+   
+<programlisting>
+List&lt;String&gt; list = new ArrayList&lt;String&gt;();
+list.add("abc");
+list.add("xyz");
+proxy.pathMultiSeparator(list);
+</programlisting>
+
+  <para>
+  and "path/separator/multi/{p}" into ".../path/separator/multi/abc-xyz". On the server side,
+  the RESTEasy runtime will turn "abc-xyz" back into a list consisting of elements "abc" and "xyz" for
+  </para>
+  
+<programlisting>
+@Path("path/separator/multi/{p}")
+@GET
+public String pathMultiSeparator(@PathParam("p") @Separator("-") List&lt;String&gt; ss) {
+   StringBuffer sb = new StringBuffer();
+   for (String s : ss) {
+      sb.append(s);
+      sb.append("|");
+   }
+   return sb.toString();
+}
+</programlisting>
+  
+   <para>
+   which will return "abc|xyz|".
+   </para>
+   
+   <para>
+   In fact, the value of the <classname>Separator</classname> annotations may be a more general regular
+   expression, which is passed to <methodname>String.split()</methodname>. For example, "[-,;]" tells the
+   server side to break up a string using either "-", ",", or ";". On the client side, a string will be created
+   using the first element, "-" in this case.
+   </para>
+   
+   <para>
+   If a parameter is annotated with <classname>@Separator</classname> with no value, then the default value is
+   </para>
+   
+   <itemizedlist>
+      <listitem>"," for a <classname>@HeaderParam</classname>, <classname>@MatrixParam</classname>,
+      <classname>@PathParam</classname>, or <classname>@QueryParam</classname>, and
+      </listitem> 
+      
+      <listitem>
+      "-" for a <classname>@CookieParam</classname>.
+      </listitem>
+   </itemizedlist>
+   
+   <para>
+   These <classname>ParamConverter</classname>s are meant to be fairly general, but there are a number of
+   restrictions:
+   </para>
+   
+   <orderedlist>
+      <listitem>They don't handle nested <classname>Collections</classname> or arrays. That is,
+      <classname>List&lt;String&gt;</classname> and <code>String[]</code> are OK, but
+      <classname>List&lt;List&lt;String&gt;&gt;</classname> and <code>String[][]</code> are not. 
+      </listitem>
+      
+      <listitem>
+      The regular expression used in <classname>Separator</classname> must match the regular expression
+<programlisting>
+"\\p{Punct}|\\[\\p{Punct}+\\]"
+</programlisting>
+
+      That is, it must be either a single instance of a punctuation symbol, i.e., a symbol in the set
+<programlisting>
+!"#$%&amp;'()*+,-./:;&lt;=&gt;?@[\]^_`{|}~
+</programlisting>
+
+      or a class of punctuation symbols like "[-,;]".
+      </listitem>

```

Comment:
We can also mention here that separator value is limited also by other specifications. Eg, "," or ";" can't be used with CookieParam because of [cookie syntax defined by RFC](https://tools.ietf.org/html/rfc6265#section-4.1.1). Another limitation is that separator can't be used in items in array/collection, (right?), eg if I have list `["a","b;c","d"]` and use separator ";", the result would be probably `["a", "b", "c", "d"]`. It is obvious, but it would be fine to have this in doc ...

Reply:
@marekkopecky, done. Thanks.